apollo-federation/src/sources/connect/json_selection/lit_expr.rs

```diff
@@ -317,6 +325,24 @@ mod tests {
             "-123.",
             LitExpr::Number(serde_json::Number::from_f64(-123.0).unwrap()),
         );
+        check_parse("00", LitExpr::Number(serde_json::Number::from(0)));
+        check_parse(
+            "-00",
+            LitExpr::Number(serde_json::Number::from_f64(-0.0).unwrap()),
+        );
```

Comment:
I guess negative zero has to be a float, since an integer wouldn't be able to represent the negativity?

Reply:
That sounds right!